### ISSUE TITLE
Missing top level heading (h1) not detecteted

### ISSUE THREAD

**Issue by tenzap** [validator/validator#2095]
### Summary

In the document, there is a H3 element.
According to spec I would have expected to have a report like "h3 present without h1"

Hopefully I didn't misunderstand the spec.

Spec: 

> If a document has one or more [headings](http://localhost:8001/#concept-heading), at least a single [heading](http://localhost:8001/#concept-heading) within the [outline](http://localhost:8001/#outline) should have a [heading level](http://localhost:8001/#heading-level) of 1.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!DOCTYPE html><html>

<head>
	<title>Title</title>
	<meta http-equiv="content-type" content="text/html;charset=utf-8">
</head>

<body>
			<h3>Heading3</h3>

</body>

</html>
```

### Checker error messages

```shell
Problem is the error is not reported.
```

### Spec links/citations

https://html.spec.whatwg.org/#headings-and-outlines-2

> If a document has one or more [headings](http://localhost:8001/#concept-heading), at least a single [heading](http://localhost:8001/#concept-heading) within the [outline](http://localhost:8001/#outline) should have a [heading level](http://localhost:8001/#heading-level) of 1.

### Steps to reproduce

1. n/a
2.
3.

### Expected behavior

n/a

### Actual behavior

n/a
